### PR TITLE
Update the log in SchemaBuilder

### PR DESCRIPTION
**Pull request by hongyunyan**
### What problem does this PR solve?

Issue Number: ref #7636

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/TiDB/Schema/SchemaSyncService.cpp [hunk 1]
```diff
@@ -210,8 +210,8 @@ bool SchemaSyncService::gc(Timestamp gc_safe_point, KeyspaceID keyspace_id)
         auto canonical_name = [&]() {
             // DB info maintenance is parallel with GC logic so we can't always assume one specific DB info's existence, thus checking its validity.
             auto db_info = tmt_context.getSchemaSyncerManager()->getDBInfoByMappedName(keyspace_id, database_name);
-            return db_info ? SchemaNameMapper().debugCanonicalName(*db_info, table_info)
-                           : "(" + database_name + ")." + SchemaNameMapper().debugTableName(table_info);
+            return db_info ? fmt::format("{}, database_id={} table_id={}", SchemaNameMapper().debugCanonicalName(*db_info, table_info), db_info->id, table_info.id)
+                           : fmt::format("({}).{}, database_id={} table_id={}", database_name, SchemaNameMapper().debugTableName(table_info), db_info->id, table_info.id);
```
Comment (JaySon-Huang) [inline dbms/src/TiDB/Schema/SchemaSyncService.cpp:214]:
```suggestion
                           : fmt::format("({}).{}, table_id={}", database_name, SchemaNameMapper().debugTableName(table_info), table_info.id);
```

Comment (JaySon-Huang) [reply dbms/src/TiDB/Schema/SchemaSyncService.cpp:214]: Fix the crash when `db_info` is nullptr

File: dbms/src/TiDB/Schema/SchemaBuilder.cpp [hunk 2]
```diff
@@ -1000,7 +1008,7 @@ void SchemaBuilder<Getter, NameMapper>::applyTable(DatabaseID database_id, Table
     {
         if (!table_info->isLogicalPartitionTable())
         {
-            LOG_ERROR(log, "new table in TiKV is not partition table {}", name_mapper.debugCanonicalName(*table_info, database_id, keyspace_id));
+            LOG_ERROR(log, "new table in TiKV is not partition table {}, database_id={}, table_id={}", name_mapper.debugCanonicalName(*table_info, database_id, keyspace_id), database_id, table_info->id);
```
Comment (JaySon-Huang) [inline dbms/src/TiDB/Schema/SchemaBuilder.cpp:1011]:
```suggestion
            LOG_ERROR(log, "new table in TiKV is not partition table {}, database_id={} table_id={}", name_mapper.debugCanonicalName(*table_info, database_id, keyspace_id), database_id, table_info->id);
```

File: dbms/src/TiDB/Schema/SchemaBuilder.cpp [hunk 3]
```diff
@@ -1032,7 +1040,7 @@ void SchemaBuilder<Getter, NameMapper>::applyTable(DatabaseID database_id, Table
     }
     else
     {
-        LOG_INFO(log, "Altering table {}", name_mapper.debugCanonicalName(*table_info, database_id, keyspace_id));
+        LOG_INFO(log, "Altering table {}, database_id={}, table_id={}", name_mapper.debugCanonicalName(*table_info, database_id, keyspace_id), database_id, table_info->id);
```
Comment (JaySon-Huang) [inline dbms/src/TiDB/Schema/SchemaBuilder.cpp:1043]:
```suggestion
        LOG_INFO(log, "Altering table {}, database_id={} table_id={}", name_mapper.debugCanonicalName(*table_info, database_id, keyspace_id), database_id, table_info->id);
```

File: dbms/src/TiDB/Schema/SchemaBuilder.cpp [hunk 4]
```diff
@@ -784,12 +790,14 @@ void SchemaBuilder<Getter, NameMapper>::applyCreatePhysicalTable(const TiDB::DBI
             if (!storage->isTombstone())
             {
                 LOG_DEBUG(log,
-                          "Trying to create table {} but it already exists and is not marked as tombstone",
-                          name_mapper.debugCanonicalName(*db_info, *table_info));
+                          "Trying to create table {} with database_id={}, table_id={}, but it already exists and is not marked as tombstone",
```
Comment (JaySon-Huang) [inline dbms/src/TiDB/Schema/SchemaBuilder.cpp:793]:
```suggestion
                          "Trying to create table {}, but it already exists and is not marked as tombstone, database_id={} table_id={}",
```

File: dbms/src/TiDB/Schema/SchemaBuilder.cpp [hunk 5]
```diff
@@ -471,19 +471,21 @@ void SchemaBuilder<Getter, NameMapper>::applyRenamePhysicalTable(
...
+        LOG_DEBUG(log, "Table {} name identical, not renaming. database_id={} table_id={}", name_mapper.debugCanonicalName(*new_db_info, new_table_info), new_db_info->id, new_table_info.id);
         return;
     }
 
     const auto old_mapped_tbl_name = storage->getTableName();
     GET_METRIC(tiflash_schema_internal_ddl_count, type_rename_table).Increment();
     LOG_INFO(
         log,
-        "Renaming table {}.{} (display name: {}) to {}.",
+        "Renaming table {}.{} (display name: {}) to {} with database_id={}, table_id={}.",
```
Comment (JaySon-Huang) [inline dbms/src/TiDB/Schema/SchemaBuilder.cpp:482]:
```suggestion
        "Renaming table {}.{} (display name: {}) to {} with database_id={}, table_id={}",
```

File: dbms/src/TiDB/Schema/SchemaBuilder.cpp [hunk 6]
```diff
@@ -538,12 +542,14 @@ void SchemaBuilder<Getter, NameMapper>::applyRecoverPhysicalTable(const TiDB::DB
         if (!storage->isTombstone())
         {
             LOG_DEBUG(log,
-                      "Trying to recover table {} but it already exists and is not marked as tombstone",
-                      name_mapper.debugCanonicalName(*db_info, *table_info));
+                      "Trying to recover table {} with database_id={}, table_id={} but it already exists and is not marked as tombstone",
```
Comment (JaySon-Huang) [inline dbms/src/TiDB/Schema/SchemaBuilder.cpp:545]:
```suggestion
                      "Trying to recover table {} but it already exists and is not marked as tombstone, database_id={} table_id={}",
```

**Review by JaySon-Huang — APPROVED** [review]

**Review by lidezhu — APPROVED** [review]

**Pull request merged**